ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -574,8 +574,8 @@ struct vk_buffer_struct {
         }
         VK_LOG_DEBUG("~vk_buffer_struct(" << buffer << ", " << size << ")");
 
-        device->device.freeMemory(device_memory);
         device->device.destroyBuffer(buffer);
+        device->device.freeMemory(device_memory);
```

Comment:
The change looks good but the commit msg should be updated to remove `wip` - would also be good to explain what specific crash this fixes in the commit msg.